framework/src/dirackernels/ReporterPointSource.C

```diff
@@ -86,31 +102,47 @@ ReporterPointSource::addPoints()
   errorCheck("weight_name", _weight.size());
   errorCheck("point_name", _point.size());
 
-  if (isParamValid("point_name"))
+  if (_read_in_points)
   {
-    for (const auto i : index_range(_values))
-    {
-      _point_to_index[_point[i]] = i;
-      addPoint(_point[i], i);
-    }
+    for (const auto & i : index_range(_point))
+      fillPoint(_point[i], i);
   }
   else
   {
+    std::vector<Point> points(_coordx.size());
     for (const auto i : index_range(_values))
     {
-      const Point pt(_coordx[i], _coordy[i], _coordz[i]);
-      _point_to_index[pt] = i;
-      addPoint(pt, i);
+      const Point point = Point(_coordx[i], _coordy[i], _coordz[i]);
+      fillPoint(point, i);
     }
   }
 }
 
 Real
 ReporterPointSource::computeQpResidual()
 {
-  //  This is negative because it's a forcing function that has been brought over to the left side
-  std::size_t index = _point_to_index[_current_point];
-  return -_test[_i][_qp] * _values[index] * _weight[index];
+  // This is negative because it's a forcing function that has been brought over to the left side
+  return -_test[_i][_qp] * _point_to_weightedValue.at(_current_point);
```

Comment:
use libmesh_map_find instead of .at
`.at` errors nastily

Reply:
thanks @GiudGiud I didn't know about that.  Its fixed now.